wicket-core/src/main/java/org/apache/wicket/core/request/handler/PageProvider.java

```diff
@@ -349,6 +306,11 @@ public void detach()
 	 */
 	public void setPageSource(IPageSource pageSource)
 	{
+		if (provision != null)
+		{
+			throw new IllegalStateException(
+				"A provision was already been done. The provider can be forcefully detached or a new one needs to be used to provide using this page source.");
```

Comment:
`"A provision has already been done. The provider can be forcefully detached or a new one has to be created be be able to use this page source."` ?